File: Challenges/450_DSA_Cracker/004_Sort_0_1_2.py
from os import *
from sys import *
from collections import *
from math import *

from sys import stdin,setrecursionlimit
setrecursionlimit(10**7)

def sort012(arr, n) :
    c0, c1, c2 = 0, 0, 0
    for i in arr : 
        if i == 0 : 
            c0 += 1
        elif i ==1 : 
            c1 += 1
        else : 
            c2 += 1
    for i in range(c0) :
        arr[i] = 0
    for i in range(c0, c0+c1) :
        arr[i] = 1
    for i in range(c0+c1, n) : 
        arr[i] = 2


#taking inpit using fast I/O
def takeInput() :
	n = int(input().strip())
	if n == 0 :
		return list(), 0
	arr = list(map(int, stdin.readline().strip().split(" ")))
	return arr, n


def printAnswer(arr, n) :
    for i in range(n) :
        print(arr[i],end=" ")
    print()
    
#main
t = int(input().strip())
for i in range(t) :
    arr, n= takeInput()
    sort012(arr, n)
    printAnswer(arr, n)

File: Challenges/450_DSA_Cracker/Array/004_Sort_0_1_2.java
import java.util.* ;
import java.io.*; 
public class Solution 
{
    public static void sort012(int[] arr)
    {
        int c0 = 0, c1 = 0, c2 = 0;
        for(int i=0; i<arr.length; i++) {
            if(arr[i] == 0) c0++;
            else if(arr[i] == 1) c1++;
            else c2++;
        }

        for(int i=0; i<c0; i++) arr[i] = 0;        
        for(int i=c0; i<c0+c1; i++) arr[i] = 1;
        for(int i=c0+c1; i<arr.length; i++) arr[i] = 2;
    }
}
